src/IntegratesQueryBuilder.js

```diff
@@ -0,0 +1,50 @@
+import { Mixin } from 'mixwith'
+
+/**
+ * @needs MakesRequests
+ */
+export default Mixin((superclass) => class extends superclass
+{
+    beforeRequest(request) {
+        if (superclass.hasOwnProperty('beforeRequest')) {
+            superclass.beforeRequest(request)
```

Comment:
Since we assume in the rest of the code that this mixin depends on `MakesRequests`, there is no need to check for the existence of `beforeRequest`. We can simply do:

```js
request = super.beforeRequest(request)
```